src/extensions/scratch3_microbit/index.js

```diff
@@ -317,7 +317,11 @@ class MicroBit {
 
         // cancel disconnect timeout and start a new one
         window.clearInterval(this._timeoutID);
-        this._timeoutID = window.setInterval(this.disconnect, BLETimeout);
+        this._timeoutID = window.setInterval(
+            // send an error to the BLE socket
+            this._ble._sendDisconnectError.bind(this._ble, 'micro:bit stopped receiving data'),
```

Comment:
This is a slight hack, but better than before.  The micro:bit extension sends an error to the BLE socket with an internal error string (that doesn't need to be translated), and then that triggers all the correct disconnect error behavior.